### PR TITLE
refactor columns

### PR DESCRIPTION
**Pull request by dtassone**

### CODE REVIEW

File: packages/grid/_modules_/grid/components/column-header-item.tsx [hunk 1]
```diff
@@ -90,9 +90,11 @@ export const ColumnHeaderItem = ({
       key={column.field}
       data-field={column.field}
       style={{
-        width,
-        minWidth: width,
-        maxWidth: width,
+        flex: column.flex == null ? `0 0 ${column.width}` : `${column.flex} ${column.flex} `,
+
```
Comment (dtassone) [inline packages/grid/_modules_/grid/components/column-header-item.tsx:94]: How about we do it this way?

File: packages/grid/_modules_/grid/hooks/features/columns/useColumns.ts [hunk 2]
```diff
@@ -168,19 +140,12 @@ export function useColumns(columns: Columns, apiRef: ApiRef): InternalColumns {
     const newState = resetState(
       columns,
       gridState.options.columnTypes,
-      viewportWidth,
+      // apiRef.current.state.containerSizes?.viewportSize.width || 0, // The fix for the old version
```
Comment (dtassone) [inline packages/grid/_modules_/grid/hooks/features/columns/useColumns.ts:143]: This is what cause the tests to fail

File: packages/grid/_modules_/grid/components/column-header-item.tsx (hunk 1 above)
Comment (DanailH) [reply packages/grid/_modules_/grid/components/column-header-item.tsx:94]: Won't this mess up the way the columns widths are calculated if it for instance you resize a column?

Comment (dtassone) [reply packages/grid/_modules_/grid/components/column-header-item.tsx:94]: Yeah I had a play, it will be difficult to manage, let's do with the width!

**Pull request closed without merging**